Library/Homebrew/env_config.rb

```diff
@@ -343,6 +343,10 @@ module EnvConfig
         default_text: "macOS: `/private/tmp`, Linux: `/tmp`.",
         default:      HOMEBREW_DEFAULT_TEMP,
       },
+      HOMEBREW_UNINSTALL_AUTOREMOVE:             {
+        description: "If set, `brew autoremove` is run after every successful call to `brew uninstall`.",
```

Comment:
Oh, I misread this before. I wonder if a `HOMEBREW_INSTALL_AUTOREMOVE` makes sense to have as well that is run on each install/upgrade/reinstall or whether a `HOMEBREW_AUTOREMOVE` should do all of these. Thoughts?

Reply:
I have no opinion on when exactly we should do this, but I have a strong opinion against having multiple environment variables.

If we do it after multiple commands then it should just be `HOMEBREW_AUTOREMOVE`.

Reply:
Works for me @carlocab. I'll personally enable `HOMEBREW_AUTOREMOVE` as soon as it's around and will primarily rely on it being used at `install` time.

Thinking about this more, though: we automatically run `brew cleanup` at install time so I wonder whether running it at `uninstall` or `cleanup` time would be the most sensible implementation of `HOMEBREW_AUTOREMOVE`?

Reply:
Running it as part of/after `brew cleanup` makes sense to me 👍

Reply:
To be clear you're suggesting running `autoremove` after `install`, `uninstall`, `upgrade` and `reinstall`, is that right?

I just took a look at how `cleanup` is handled internally and it makes me think that a separate `autoremove.rb` file might be the way to go with this.

Also, `HOMEBREW_AUTOREMOVE` makes sense to me too.